src/tests/test_index.py

```diff
@@ -58,4 +65,52 @@ def test_index_abstract(tmp_path):
     query = the_index.query_index("test_test")
     assert len(query) == 0
 
-    assert the_index.n_articles() == 2
\ No newline at end of file
+    assert the_index.n_articles() == 2
+
+def test_citation_count(tmp_path):
+    assert not os.path.exists(util.get_index_dir(tmp_path))
+
+    cataloger = AbstractCatalog(tmp_path)
+    abs1 = Abstract(1000, 2020, "A Really Cool Pubmed Abstract",
+        "test The quick brown fox jumped over the lazy dog repeated the repeat.")
+    abs2 = Abstract(1001, 2021, "A Cool Title",
+        "Some of the words are are the repeated repeat but some are-are-are not.")
+    abs3 = Abstract(1002, 2022, "sdfsb  rgtd gfhdfg",
+        "Test repeat test repeat")
+    ct = {1000:1, 1001:2, 1002:3}
+    with open(str(tmp_path) + "/test.json", "w", encoding="utf-8") as f:
+            json.dump(ct, f)
+    cataloger.add_or_update_abstract(abs1)
+    cataloger.add_or_update_abstract(abs2)
+    cataloger.add_or_update_abstract(abs3)
+    cataloger.write_catalog_to_disk(util.get_abstract_catalog(tmp_path))
+
+    indexer = IndexBuilder(tmp_path)
+    hot_storage = dict()
+    cold_storage = dict()
+    indexer._index_abstract(abs1, hot_storage)
+    indexer._serialize_hot_to_cold_storage(hot_storage, cold_storage)
+    indexer._write_index_to_disk(cold_storage)
+
+    indexer._index_abstract(abs2, hot_storage)
+    indexer._serialize_hot_to_cold_storage(hot_storage, cold_storage, True)
+    indexer._write_index_to_disk(cold_storage)
+
+    indexer._index_abstract(abs3, hot_storage)
+    indexer._serialize_hot_to_cold_storage(hot_storage, cold_storage, True)
+    indexer._write_index_to_disk(cold_storage)
```

Comment:
you can call `_index_abstract()` on the 3 abstracts, then write to disk. don't need to do serialize/write after each abstract